$NetBSD: patch-asmcomp_amd64_emit.mlp,v 1.5 2020/01/14 19:53:35 jaapb Exp $

Fix attributes and entry size or the .rodata mergable constant section

--- asmcomp/amd64/emit.mlp.orig	2018-07-17 21:17:05.536787953 +0000
+++ asmcomp/amd64/emit.mlp
@@ -986,7 +986,7 @@ let begin_assembly() =
     | S_macosx -> D.section ["__TEXT";"__literal16"] None ["16byte_literals"]
     | S_mingw64 | S_cygwin -> D.section [".rdata"] (Some "dr") []
     | S_win64 -> D.data ()
-    | _ -> D.section [".rodata.cst8"] (Some "a") ["@progbits"]
+    | _ -> D.section [".rodata.cst8"] (Some "aM") ["@progbits,8"]
     end;
     D.align 16;
     _label (emit_symbol "caml_negf_mask");
@@ -1046,7 +1046,7 @@ let end_assembly() =
     | S_macosx -> D.section ["__TEXT";"__literal8"] None ["8byte_literals"]
     | S_mingw64 | S_cygwin -> D.section [".rdata"] (Some "dr") []
     | S_win64 -> D.data ()
-    | _ -> D.section [".rodata.cst8"] (Some "a") ["@progbits"]
+    | _ -> D.section [".rodata.cst8"] (Some "aM") ["@progbits,8"]
     end;
     List.iter (fun (cst,lbl) -> emit_float_constant cst lbl) !float_constants
   end;
